pyrk/th_component.py

```diff
@@ -83,33 +131,232 @@ def update_temp(self, timestep, temp):
+        :param k: thermal conductivity of the material that heat is conducted in(may not be the 'self' component)
+        :type k:float
+        :param area: conduction surface for the slab
+        :type area: float
+        :param L: thickness of the slab
+        :type L: float
+        :param r_b: outer radius of the component
+        :type r_b: float
+        :param r_env: outer radius of the environment
+        :type r_env: float
+        '''
+        self.cond[env] = {
+            "k": k.to('watts/meter/kelvin'),
+            "area": area.to('meter**2'),
+            "L": L.to('meter'),
+            "r_b": r_b.to('meter'),
+            "r_env": r_env.to('meter')
+        }
+
+    def add_advection(self, name, m_flow, t_in, cp):
+        '''Add advection dictionary to the fluid component(coolant) that has
+        advective heat tranfer
+
+        :param m_flow: mass flow rate
+        :type m_flow: float
+        :param t_in: temperature at the inlet of the control volume
+        :type t_in: float
+        :param cp: specific heat capacity
+        :type cp: float
+        '''
+        self.adv[name] = {
+            "m_flow": m_flow.to('kg/second'),
+            "t_in": t_in.to('kelvin'),
+            "cp": cp.to('joule/kg/kelvin')
+        }
+
+
+class THSuperComponent(object):
+
+    '''A 'component' containing a list of component
+    Creating a superComponent would automatically define conduction between the
+    mesh elements'''
+
+    def __init__(self, name, T0, sub_comp=[], timer=Timer()):
+
+        """Initalizes a thermal hydraulic super component.
+
+        :param name: The name of the supercomponent (i.e., "fuel" or "cool")
+        :type name: str.
+        :param T0: The initial temperature of the supercomponent
+        :type T0: float.
+        :param sub_comp: List of components that makes up the supercomponent.
+        The sub_components should be in order from the center to the outside
+        :type sub_comp: list of THComponent
+        :param timer: The timer instance for the sim
+        :type timer: Timer object
+        """
+        self.name = name
+        self.T0 = T0
+        self.sub_comp = sub_comp
+        self.timer = timer
+        self.T = units.Quantity(np.zeros(shape=(timer.timesteps(),),
+                                         dtype=float), 'kelvin')
+        self.T[0] = T0
+        self.conv = {}
+        self.add_conduction_in_mesh()
+        self.alpha_temp = 0.0*units.delta_k/units.kelvin
+
+    def dtemp(self, t_idx, t_feedback_idx):
+        '''calculate temperature different between two timesteps
+        :param t_idx: the timestep at which to calculate reactivity feedback
+        :type t_idx: int
+        :param t_feedback_idx: the timestep at which the temperature is used as
+        reference temperature
+        :type t_feedback_idx: int
+        '''
+        T0 = self.T[t_feedback_idx]
+        return self.T[t_idx-1]-T0
+
+    def temp_reactivity(self, t_idx, t_feedback_idx):
```

Comment:
This also should be inherited, rather than copy-pasted from the function that already exists in THComponent.